RecoTracker/LSTCore/src/alpaka/Quintuplet.h

```diff
@@ -1710,107 +1729,146 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE::lst {
+        }
+
+        // Step 2: Parallel processing of triplet pairs
+        for (int i = flatThreadIdxXY; i < matchCount; i += flatThreadExtent) {
+          unsigned int quintupletIndex = ranges.quintupletModuleIndices()[lowerModule1] + i;
+          int innerTripletIndex = quintuplets.PreAllocatedTripletIndices()[quintupletIndex][0];
+          int outerTripletIndex = quintuplets.PreAllocatedTripletIndices()[quintupletIndex][1];
+
+          uint16_t lowerModule2 = triplets.lowerModuleIndices()[innerTripletIndex][1];
+          uint16_t lowerModule3 = triplets.lowerModuleIndices()[innerTripletIndex][2];
+          uint16_t lowerModule4 = triplets.lowerModuleIndices()[outerTripletIndex][1];
+          uint16_t lowerModule5 = triplets.lowerModuleIndices()[outerTripletIndex][2];
+
+          float innerRadius, outerRadius, bridgeRadius, regressionCenterX, regressionCenterY, regressionRadius,
+              rzChiSquared, chiSquared, nonAnchorChiSquared, dBeta1, dBeta2;  //required for making distributions
+
+          float t5Embed[Params_T5::kEmbed] = {0.f};
+
+          bool tightCutFlag = false;
+
+          bool success = runQuintupletDefaultAlgo(acc,
+                                                  modules,
+                                                  mds,
+                                                  segments,
+                                                  triplets,
+                                                  lowerModule1,
+                                                  lowerModule2,
+                                                  lowerModule3,
+                                                  lowerModule4,
+                                                  lowerModule5,
+                                                  innerTripletIndex,
+                                                  outerTripletIndex,
+                                                  innerRadius,
+                                                  outerRadius,
+                                                  bridgeRadius,
+                                                  regressionCenterX,
+                                                  regressionCenterY,
+                                                  regressionRadius,
+                                                  rzChiSquared,
+                                                  chiSquared,
+                                                  nonAnchorChiSquared,
+                                                  dBeta1,
+                                                  dBeta2,
+                                                  tightCutFlag,
+                                                  t5Embed,
+                                                  ptCut);
+          if (success) {
+            int totOccupancyQuintuplets = alpaka::atomicAdd(
+                acc, &quintupletsOccupancy.totOccupancyQuintuplets()[lowerModule1], 1u, alpaka::hierarchy::Threads{});
```

Comment:
After seeing the comment on my open PR, I'm not sure if this atomic add with hierarchy threads is safe? Since you spread all potential pairs over all threads, do these atomic adds need to be changed to alpaka::hierarchy::Blocks{}?

Reply:
I think the hierarchy on the thread level is safe as long as different blocks are not touching the same memory space for atomic add, and in this case, each module is a different block, so the totOccupancyQuintuplets per module is counted per block and still fine.